mozilla/bedrock
# 11022: Support multi-locale content from Contentful in the VPN Resource Center

_Please note: this PR includes all of #11002, so you may want to review that first and merge it to mozilla/master to reduce the diff size here_ 

## Description
This changeset builds on the work in #11002 (sync down Contentful content in multiple locales) by actually displaying it in the VRC in a way that fits with exisitng Bedrock behaviour

* Content for which we have a specific loclized version is available if requested.
  * The locale switcher in the footer is populated with the appropriate set of locales that content is available in (be it the VRC listing page or an article page.

* We will consider a locale to be active when there is at least certain, configurable percentage of content available in that locale compared to the default en-US locale. If the percentage is below the threshold, we act like the content is not yet in the system.  
  * The default threshold percentage for production is currently set at 60% - this was just a gut decision to get us moving
  * When `DEV=True` the threshold percentage is set very low - 1% - so that content in any published locale is shown pretty much all the time. We can experiment to see if this is has any rough corners IRL.

## Issue / Bugzilla link

Resolves #11022 

## Testing

* Set `CONTENTFUL_ENVIRONMENT=sandbox` in your  `.env` because I've set up some small additional-locale examples in there.
* Restart your `app` container
* `make shell` 
* `./manage.py update_contentful --force`
* Locally, visit the VRC at http://localhost:8080/en-US/products/vpn/resource-center/
* Check the locale switcher in the footer - confirm you can switch to German (two fake-example articles) and French (one fake-example article)
* Explore those links on the DE and FR pages- confirm you don't get booted back to the default en-US
* Try changing the slug so that a page that is available becomes not available due to its locale - eg set the locale to `ru` or `ja` or `es-MX` - you should get 404ed
* Try changing the slug for a known locale (eg `de`) to one that goes not exist. You will get a 404
* Try changing the locale for a known page to a different locale which lacks a translation for that page - eg change http://localhost:8080/de/products/vpn/resource-center/the-difference-between-a-vpn-and-a-web-proxy/ to a `fr` locale. You should get sent to the default `en-US` locale page, because we do have content for that slug in that fallback locale
